floodlight/io/statsperform.py

```diff
@@ -427,3 +432,683 @@ def read_open_statsperform_tracking_data_csv(
+            [number_of_frames[segment], number_of_away_players * 2], np.nan
+        )
+        for segment in segments
+    }
+    xydata["Ball"] = {
+        segment: np.full([number_of_frames[segment], 2], np.nan) for segment in segments
+    }
+
+    # read txt file from disk
+    with open(filepath_txt, "r") as f:
+        tracking_data_lines = f.readlines()
+
+    # loop
+    for package in tracking_data_lines:
+
+        # read line to get gameclock, player positions and ball info
+        (
+            gameclock,
+            segment,
+            positions,
+            ball,
+        ) = _read_full_tracking_data_txt_single_line(package)
+
+        # check if frame is in any segment
+        if segment is None:
+            # skip line if not
+            continue
+        else:
+            # otherwise calculate relative frame (in respective segment)
+            frame_rel = int((gameclock - periods[segment][0]) / framerate)
+
+        # insert (x,y)-data into np.array
+        for team in ["Home", "Away"]:
+            for jID in positions[team].keys():
+                # map jersey number to array index and infer respective columns
+                x_col = (links[team][jID] - 1) * 2
+                y_col = (links[team][jID] - 1) * 2 + 1
+                xydata[team][segment][frame_rel, x_col] = positions[team][jID][0]
+                xydata[team][segment][frame_rel, y_col] = positions[team][jID][1]
+
+        # get ball data
+        xydata["Ball"][segment][frame_rel] = ball.get("position", np.nan)
+
+    # create XY objects
+    home_ht1 = XY(xy=xydata["Home"][1], framerate=framerate)
+    home_ht2 = XY(xy=xydata["Home"][2], framerate=framerate)
+    away_ht1 = XY(xy=xydata["Away"][1], framerate=framerate)
+    away_ht2 = XY(xy=xydata["Away"][2], framerate=framerate)
+    ball_ht1 = XY(xy=xydata["Ball"][1], framerate=framerate)
+    ball_ht2 = XY(xy=xydata["Ball"][2], framerate=framerate)
+
+    data_objects = (home_ht1, home_ht2, away_ht1, away_ht2, ball_ht1, ball_ht2)
+
+    return data_objects
+
+
+def read_event_data_from_url(
+    url_events: str,
+) -> Tuple[Events, Events, Events, Events, Pitch]:
+    """Reads a URL from the StatsPerform API (StatsEdgeViewer) containing an events csv
+    file and extracts the stored event data and pitch information.
+
+    Parameters
+    ----------
+    url_events: str
+        URL to the xml file containing the event data.
+
+    Returns
+    -------
+    data_objects: Tuple[Events, Events, Events, Events, Pitch]
+        Events-objects for both teams and both halves and pitch information. The order
+        is (home_ht1, home_ht2, away_ht1, away_ht2, pitch).
+    """
+    data_dir = os.path.join(DATA_DIR, "statsperform")
+    temp_file = "events_temp.xml"
+    download_event_data_from_url(url_events=url_events, filename_xml=temp_file)
+    data_objects = read_event_data_xml(filepath_xml=os.path.join(data_dir, temp_file))
+    os.remove(os.path.join(data_dir, temp_file))
```

Comment:
Have you tested this with other operating systems?

Reply:
Hadn't had the opportunity yet unfortunately...